tests/scripts/all.sh

```diff
@@ -122,6 +122,10 @@ pre_initialize_variables () {
     FORCE=0
     KEEP_GOING=0
 
+    # Seed value used with the --release-test option.
+    # !!! Keep this in sync with SEED in basic-build-test.sh !!!
+    RELEASE_SEED=1
```

Comment:
Do we really need to keep them in sync? Would it be a problem if the seed for ssl-opt.sh were different when called through all.sh or basic-build-test.sh? It would still be deterministic.

Reply:
The seed determines the randomization that udp_proxy does to drop/duplicate/delay packets. The advantage of having the same seed is that you can debug a test case without worrying about how it was called.

Reply:
ok, it makes sense, could we add a comment explaining this?

Reply:
Thinking about it, why not putting the default value in ssh-opt.sh?

Reply:
IIRC the idea of making `ssl-opt.sh` random by default was to have more diversity when developers run tests. But it's deterministic on CI so that CI is repeatable.

In the long term, I think udp_proxy should not be randomized, and we should make sure that each test case does what it's supposed to do. For example (from memory, details might be slightly different), there's a test case that's supposed to test dropping packets, but ~1% of the seeds drop zero packets which causes the test to fail even though the library behaved correctly; instead that test case should forcibly drop a packet in a deterministic way.

Reply:
Yes, historically `ssl-opt.sh` was random by default for diversity. Then experience showed it didn't work well of the CI and we added an option to make it deterministic, either by setting `SEED` in the environment, or using `--seed` on the command line.

Now I'm thinking it may make sense to turn things around, and instead make it deterministic by default, and have an option to make it random, for example `--random-seed` on the command line or `SEED` set to the empty string (or a special string such as "random") in the environment.

That way, `basic-build-test.sh` wouldn't have to do anything, `all.sh -r` wouldn't have to do anything, and `all.sh --random-seed` would set `SEED` in the environment to the empty (or special) string that signals to `ssl-opt.sh` to no use a fixed seed, and `all.sh --seed <value>` would set `SEED=<value>` in the environment to be used by `ssl-opt.sh`.

Wdyt?

Reply:
Long-term, I would like SSL testing to be deterministic, with each test case actually doing what it says on the tin. But short-term, I could warm up to your proposal. In the very short term, though, I'd prefer to merge this PR so that we can take our time to discuss it.

Reply:
I tend to agree. @ronald-cron-arm would you agree with this PR as a hotfix for the frequent failures of the nightly job? Then as Gilles said we could have more time to discuss a better solution.

Reply:
Yes fine by me. Maybe we could remove "# !!! Keep this in sync with SEED in basic-build-test.sh !!!" because it seems that eventually the sync is not that important, just more convenient to debug as Gilles mentioned.